.azure-pipelines/templates/platform_pr_pipeline.yml

```diff
@@ -24,4 +25,22 @@ stages:
           script_parameter: |
             python -m mlops.${{ parameters.model_type }}.src.mlops_pipeline \
               --build_environment ${{parameters.exec_environment}} \
-              --wait_for_completion True
+              --wait_for_completion True \
+              --output_file run_id.txt
+      - task: AzureCLI@2
+        displayName: "Cancel AML Pipeline"
+        condition: canceled()
+        inputs:
+          azureSubscription: $(AZURE_RM_SVC_CONNECTION)
+          scriptType: bash
+          scriptLocation: inlineScript
+          inlineScript: |
+            echo "Pipeline was canceled. Attempting to cancel AML Pipeline..."
+
+            readarray arr <"run_id.txt"
+            run_name=${arr[0]}
+
+            az extension add --name ml
+            az ml job cancel --name $run_name \
+              --resource-group $(RESOURCE_GROUP_NAME) \
+              --workspace-name $(WORKSPACE_NAME)
```

Comment:
Maybe add an echo statement so we know the job was successfully canceled?
```
                az ml job cancel --name $run_name \
                  --resource-group $(RESOURCE_GROUP_NAME) \
                  --workspace-name $(WORKSPACE_NAME) && echo "Successfully canceled the AML Pipeline."
```

Reply:
![image](https://github.com/user-attachments/assets/3017d513-8cef-4718-8beb-5c0734431b52)